pkg/rangemap/rm_test.go

```diff
@@ -0,0 +1,138 @@
+package rangemap
+
+import (
+	"encoding/binary"
+	"github.com/google/go-cmp/cmp"
+	"testing"
+)
+
+type TestRange struct {
+	KeyStart []byte
+	KeyEnd   []byte
+	Value    string
+}
+
+func (t TestRange) Start() []byte {
+	return t.KeyStart
+}
+
+func (t TestRange) End() []byte {
+	return t.KeyEnd
+}
+
+type KeyVal struct {
+	key   []byte
+	value string
+}
+
+type Test struct {
+	ranges              []TestRange
+	invalidRanges       []TestRange
+	keyValues           []KeyVal
+	removeValidRanges   []TestRange
+	removeInvalidRanges []TestRange
+}
+
+func getMaxKey() []byte {
+	i := int64(1024)
+	res := make([]byte, 64)
+	binary.PutVarint(res, i)
+	return res
+}
+
+func prepareTests() map[string]Test {
+	maxKey := getMaxKey()
+	return map[string]Test{
+		"simple": {
+			ranges: []TestRange{
+				{[]byte("a"), []byte("o"), "H1"},
```

Comment:
You can keep the strings here and convert to byte array below

Reply:
doen